solr/core/src/test/org/apache/solr/client/solrj/embedded/TestEmbeddedSolrServerConstructors.java

```diff
@@ -58,4 +63,58 @@ public void testNodeConfigConstructor() throws Exception {
       assertEquals(1, server.query("newcore", new SolrQuery("*:*")).getResults().getNumFound());
     }
   }
+
+  @Test
+  public void testPathConstructorZipFS() throws Exception {
+    Path dataDir = createTempDir("data-dir");
+    Path archive = createTempFile("configset", ".zip");
+    Files.delete(archive);
+
+    // When :
+    // Prepare a zip archive which contains
+    // the configset files as shown below:
+    //
+    // configset.zip
+    // └── 1
+    //     └── 2
+    //         └── 3
+    //             └── 4
+    //                 ├── data
+    //                 │   └── core1
+    //                 │       ├── conf
+    //                 │       │   ├── managed-schema.xml
```

Comment:
BTW if you actually did want to launch Solr from a ZIP; the choice of a "managed" schema is dubious as it implies the mutability of the schema, yet the zip is read-only.  In my experience, I've only seen people actually use a managed schema as-intended for POCs.  I suppose you worked around this with the memory based StorageIO thing.